addons/html_builder/static/src/plugins/image/image_grid_option_plugin.js

```diff
@@ -0,0 +1,45 @@
+import { Plugin } from "@html_editor/plugin";
+import { registry } from "@web/core/registry";
+import { ImageGridOption } from "./image_grid_option";
+
+class ImageGridOptionPlugin extends Plugin {
+    static id = "imageGridOption";
+
+    resources = {
+        builder_options: [
+            {
+                OptionComponent: ImageGridOption,
+                selector: "img",
+            },
+        ],
+        builder_actions: this.getActions(),
+    };
+
+    getActions() {
+        return {
+            setGridImageMode: {
+                isApplied: ({ editingElement, value: className }) => {
+                    const imageGridItemEl = editingElement.closest(".o_grid_item_image");
+                    if (
+                        className === "" &&
+                        !imageGridItemEl.classList.contains("o_grid_item_image_contain")
+                    ) {
+                        return true;
+                    } else {
+                        return imageGridItemEl.classList.contains(className);
+                    }
+                },
+                apply: ({ editingElement, value: className }) => {
+                    const imageGridItemEl = editingElement.closest(".o_grid_item_image");
+                    if (className === "o_grid_item_image_contain") {
+                        imageGridItemEl.classList.add("o_grid_item_image_contain");
+                    } else {
+                        imageGridItemEl.classList.remove("o_grid_item_image_contain");
+                    }
```

Comment:
Why not use "cover" and "contains" value ? like before ?

Reply:
I didn't like the way they did before cause it's only with or without this class :grimacing: if we just pass the class name, I think the `isApplied` function is a bit simpler :3 (I should have replaced the class name in the first if)